src/App.js

```diff
@@ -32,13 +52,64 @@ function App() {
     }
   }, [todoList, isLoading]);
 
-  function addTodo(newTodo) {
-    setTodoList((prevState) => [...prevState, newTodo]);
-  }
+  // to add an item to Airtable
+  const addTodo = async (todo) => {
+    const airtableData = {
+      fields: {
+        title: todo,
+      },
+    };
 
-  const removeTodo = (id) => {
-    const newList = todoList.filter((item) => item.id !== id);
-    setTodoList(newList);
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+      body: JSON.stringify(airtableData),
+    };
+
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+        throw new Error(message);
+      }
+
+      const resp = await response.json();
+      const newTodo = { id: resp.id, title: resp.fields.title };
+      setTodoList([...todoList, newTodo]);
```

Comment:
Let me share with you another way to handle the rendering of the todos after the addTodo and removeTodo functions:

```
const postData = async (newTodo) => {
try {
      const todoData = {
        fields: newTodo,
      };

      const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;

      const options = {
        body: JSON.stringify(todoData),
        method: "POST",
        headers: {
          "Content-type": "application/json",
          Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
        },
      };

      const response = await fetch(url, options);
      if (!response.ok) {
        throw new Error(`Error: ${response.status}`);
      }
      const data = await response.json();
      // Call fetchData to refresh the todo list
      fetchData();
    } catch (error) {
      console.log(error);
    }
  };
```

The following code:
```
// Call fetchData to refresh the todo list
fetchData();
```
would be similar for the removeTodo function.
With that line you'll be getting the todos from the server instead of setting them up with setTodos in the client.

Both approaches have their pros and cons, and the best one depends on your specific use case.

**Fetching data after each operation (my approach):**

Pros:

The data on the client is always synchronized with the server. This is beneficial if the data on the server can be changed by other users or processes.
Cons:

It can lead to performance issues if the amount of data is large, as you're fetching all data after each operation.


**Updating local state after each operation (your approach):**

Pros:

It's more performant for large amounts of data, as you're only updating the local state, not fetching all data from the server.
Cons:

The data on the client can become out of sync with the server if the server data can be changed by other users or processes.

In general, if the data on the server is not expected to change by other users or processes (like for this homework), updating the local state after each operation can be more performant. However, if the server data can be changed by other users or processes, fetching data after each operation can help keep the client data synchronized with the server.